### ISSUE TITLE
Report Generation Broken 

### ISSUE THREAD

**Issue by jhughesbiot** [faucetsdn/daq#618]
Not sure what recent update introduced this error but DAQ can no longer complete and generate reports.  I recently updated and merged the latest commits to my branch today and now every time I run cmd/run -s it runs continually because of the below error.  This error did not exist earlier today, just after this merge. I have rebuild the entire environment and deleted the inst directory but it is always met with this error.  

Aug 26 02:30:50 runner   ERROR   result already exists
Traceback (most recent call last):
  File "/home/ubuntu/jhughesbiot/daq/daq/runner.py", line 364, in _handle_system_idle
    self.target_set_complete(device, 'target set not active')
  File "/home/ubuntu/jhughesbiot/daq/daq/runner.py", line 692, in target_set_complete
    self._target_set_cancel(device)
  File "/home/ubuntu/jhughesbiot/daq/daq/runner.py", line 729, in _target_set_cancel
    target_host.terminate('_target_set_cancel', trigger=False)
  File "/home/ubuntu/jhughesbiot/daq/daq/host.py", line 422, in terminate
    self._finalize_report()
  File "/home/ubuntu/jhughesbiot/daq/daq/host.py", line 405, in _finalize_report
    report_paths = self._report.finalize()
  File "/home/ubuntu/jhughesbiot/daq/daq/report.py", line 147, in finalize
    self._process_results()
  File "/home/ubuntu/jhughesbiot/daq/daq/report.py", line 180, in _process_results
    self._accumulate_result(test_name, result, extra, module_name=module_name)
  File "/home/ubuntu/jhughesbiot/daq/daq/report.py", line 212, in _accumulate_result
    assert test_name not in self._results, 'result already exists'

**Comment by henry54809** [comment]
Hi Jonathan, the assertion error means the result for test name exists already in the report; this likely was introduced was in the merging, would you please check the diff between origin master and your branch.

**Comment by grafnu** [comment]
and/or include the bin/techsupport, it should have a bunch of logs and
other things that would clarify. The interesting stuff isn't just in the
stack-trace.

**Comment by jhughesbiot** [comment]
If it is a duplicate test name that's the error, it's pretty much impossible to know that from that failure since the report never gets created to compare so you would never know how to debug this issue or resolve which test is the issue.

I just rebuilt everything directly from this repo, not my branch and the issue didn't present itself so must be something else I changed in the working branch at the same time to make this show up.  

Ran the test_module with the module I was working on and can see the duplicate in the output.  Not sure where it came from but this may be an indicator of a specific error report that could be used to identify the exact duplicate for troubleshooting.
